packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -112,7 +112,42 @@ export default class OperatorModeStateService extends Service {
     for (let item of stackItems) {
       if ('card' in item && item.card.id == id) {
         let document = await this.cardService.serializeCard(item.card);
-        document.data = merge(document.data, patch);
+        if (patch.attributes) {
+          for (let key of Object.keys(patch.attributes)) {
+            if (!(key in item.card)) {
+              throw new Error(`Attribute "${key}" does not exist in card`);
+            }
+          }
+          document.data.attributes = merge(
```

Comment:
i don't know if this necessarily does the right thing--for instance if the ai bot is trying to remove a property in the patch, does `merge()` actually remove it?

Reply:
take a look at how we handle patches on the server in `Realm.patchCard()`, we use `mergeWith()` which i think is probably what we want here. (the test case is actually patching arrays)